File: output/GPT4o-Mini/xCodeEval_remove_4000/C++/C/57ed27e3434318516db4a77e0f00b859.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

#define REP(i, n) for (int i = 0; i < (n); ++i)
#define SZ(x) ((int)(x).size())

typedef long long LL;

char str[1200];
int use[1200];

int main() {
    scanf("%s", str);
    LL res = 1;
    memset(use, 0, sizeof(use));
    LL zero = 0, all = 10;
    int n = strlen(str);

    REP(i, n) {
        if (i == 0) {
            if (str[i] >= 'A' && str[i] <= 'J') {
                res *= 9;
                all--;
                use[str[i]] = 1;
            } else if (str[i] == '?') {
                res *= 9;
            }
        } else {
            if (str[i] >= 'A' && str[i] <= 'J') {
                if (use[str[i]]) continue;
                use[str[i]] = 1;
                res *= all;
                all--;
            } else if (str[i] == '?') {
                ++zero;
            }
        }
    }
    
    printf("%lld", res);
    while (zero--) putchar('0');
    puts("");
    
    return 0;
}

File: output/GPT4o-Mini/xCodeEval_remove_4000/C++/Python/57ed27e3434318516db4a77e0f00b859.py
from collections import defaultdict

def main():
    str_input = input().strip()
    res = 1
    use = [False] * 1200
    zero = 0
    all = 10
    n = len(str_input)

    for i in range(n):
        if i == 0:
            if 'A' <= str_input[i] <= 'J':
                res *= 9
                all -= 1
                use[ord(str_input[i])] = True
            elif str_input[i] == '?':
                res *= 9
        else:
            if 'A' <= str_input[i] <= 'J':
                if use[ord(str_input[i])]:
                    continue
                use[ord(str_input[i])] = True
                res *= all
                all -= 1
            elif str_input[i] == '?':
                zero += 1

    print(res, end='')
    print('0' * zero)

main()
